src/samplers/independent.cpp

```diff
@@ -28,63 +28,43 @@ by Melissa O’Neill.
 
 This is the most basic sample generator; because no precautions are taken to avoid
 sample clumping, images produced using this plugin will usually take longer to converge.
-In theory, this sampler is initialized using a deterministic procedure, which means
-that subsequent runs of Mitsuba should create the same image. In practice, when
-rendering with multiple threads and/or machines, this is not true anymore, since the
-ordering of samples is influenced by the operating system scheduler.
+this sampler is initialized using a deterministic procedure, which means that subsequent runs
+of Mitsuba should create the same image.
+
+.. subfigstart::
+.. subfigure:: ../../resources/data/docs/images/sampler/independent_1024_samples.svg
+   :caption: 1024 samples projected onto the first two dimensions.
+.. subfigure:: ../../resources/data/docs/images/sampler/independent_64_samples_and_proj.svg
+   :caption: 64 samples projected onto the first two dimensions and their
+             projection on both 1D axis (top and right plot). Both the 2D and 1D projections
+             exhibit sample clumping which will result in higher variance in the rendered image.
+.. subfigend::
+   :label: fig-independent-pattern
 
  */
 
 template <typename Float, typename Spectrum>
-class IndependentSampler final : public Sampler<Float, Spectrum> {
+class IndependentSampler final : public PCG32Sampler<Float, Spectrum> {
 public:
-    MTS_IMPORT_BASE(Sampler, m_sample_count, m_base_seed)
+    MTS_IMPORT_BASE(PCG32Sampler, m_sample_count, m_base_seed, m_rng, seed)
     MTS_IMPORT_TYPES()
 
-    using PCG32 = mitsuba::PCG32<UInt32>;
-
     IndependentSampler(const Properties &props = Properties()) : Base(props) {
         /* Can't seed yet on the GPU because we don't know yet
            how many entries will be needed. */
         if (!is_dynamic_array_v<Float>)
-            seed(PCG32_DEFAULT_STATE);
+            seed(PCG32_DEFAULT_STATE, 1);
     }
 
-    ref<Base> clone() override {
+    ref<Sampler<Float, Spectrum>> clone() override {
         IndependentSampler *sampler = new IndependentSampler();
         sampler->m_sample_count = m_sample_count;
         sampler->m_base_seed = m_base_seed;
         return sampler;
     }
 
-    /// Seeds the RNG with the specified size, if applicable
-    void seed(UInt64 seed_value) override {
-        if (!m_rng)
-            m_rng = std::make_unique<PCG32>();
-
-        seed_value += m_base_seed;
-
-        if constexpr (is_dynamic_array_v<Float>) {
-            UInt64 idx = arange<UInt64>(seed_value.size());
-            m_rng->seed(sample_tea_64(seed_value, idx),
-                        sample_tea_64(idx, seed_value));
-        } else {
-            m_rng->seed(seed_value, PCG32_DEFAULT_STREAM + arange<UInt64>());
-        }
-    }
-
     Float next_1d(Mask active = true) override {
-        if constexpr (is_dynamic_array_v<Float>) {
-            if (m_rng == nullptr)
```

Comment:
Are these kinds of error checks still somewhere in the code? They were quite helpful to debug sampler-related bugs.

Reply:
`m_rng` isn't a pointer anymore so I removed that check. If you would like to add a assertion whether the sampler was seeded, we can check that `m_wavefront_size != 0`.

Reply:
Yeah, that sounds good.